### DISCUSSION TITLE
Is custom diff tool supported?

### DISCUSSION THREAD

**Discussion by roastduck** [airblade/vim-gitgutter#856] [Q&A] [answered]
Hi,

I have configured git to use a custom diff tool other than the system `diff`, by configurate in `~/.gitconfig` to use [`difft`](https://github.com/Wilfred/difftastic).

```
[diff]
        external = difft
```

However, it seems vim-gitgutter is not aware of this configuration. Does vim-gitgutter follow `~/.gitconfig`? If not, can I configure the diff tool explicitly?

**Comment by airblade** [comment, marked as answer]
You deliberately can't configure the diff tool because only plain git-diff makes sense for gitgutter.  GitGutter uses the diff tool to get the patch headers, apply patches, etc.

difftastic's first ["non-goal"](https://github.com/Wilfred/difftastic#non-goals) is patching.  It says:

> Difftastic output is intended for human consumption, and it does not generate patches that you can apply later. Use `diff` if you need a patch.
